lexers/g/gdscript.go

```diff
@@ -121,4 +130,21 @@ var GDScript = internal.Register(MustNewLexer(
 			{`\n`, LiteralStringSingle, nil},
 		},
 	},
-))
+).SetAnalyser(func(text string) float32 {
+	var result float32 = 0
```

Comment:
If we use `float64` here, you do not need to convert below.
 ```suggestion
	var result float64 = 0
```